core-bundle/src/Resources/contao/library/Contao/Files.php

```diff
@@ -156,6 +156,11 @@ public function fopen($strFile, $strMode)
 	{
 		$this->validate($strFile);
 
+		if ($strMode[0] !== 'r')
```

Comment:
```suggestion
		if ($strMode[0] !== 'r' && \dirname($strFile) && \dirname($strFile) !== '.')
```

Otherwise files from the root directory could cause problems I think `->fopen('foo.txt')`.
_(A reason why I’d prefer using Symfony Filesystem ☺️)_

Reply:
what would that "attack" look like? `./file.php`?

Reply:
also, wouldn't that be catched by `is_dir` in `mkdir`?

Reply:
`Files::getInstance()->fopen('test.txt', 'w')` results in `$this->mkdir('.')` and `'.'` is an insecure path (I hope).

Reply:
changed in 8209e66dbc84d36c41db05dafebb797a00980c55

Reply:
changed in 8209e66dbc84d36c41db05dafebb797a00980c55